lib/engine/game/g_1873/game.rb

```diff
@@ -3147,6 +3151,276 @@ def game_hexes
+                   'frame=color:#800080',
+              %w[
+                B13
+              ] => 'city=slots:2,revenue:yellow_30|green_70|brown_60|gray_60;'\
+                   'path=a:4,b:_0,track:narrow;path=a:5,b:_0,track:narrow;'\
+                   'frame=color:#800080;icon=image:1873/ZW_open,sticky:1,large:1',
+              %w[
+                C4
+              ] => 'city=revenue:yellow_50|green_80|brown_120|gray_150;path=a:5,b:_0,track:narrow',
+              %w[
+                C6
+              ] => 'town=revenue:0;path=a:0,b:_0,track:narrow;'\
+                   'icon=image:1873/SBC6_open,sticky:1,large:1',
+              %w[
+                E18
+              ] => 'city=revenue:30,slots:2;path=a:1,b:_0,track:narrow;'\
+                   'path=a:2,b:_0,track:narrow;path=a:4,b:_0,track:narrow;'\
+                   'icon=image:1873/PM_open,sticky:1,large:1',
+              %w[
+                F15
+              ] => 'city=revenue:yellow_30|green_40|brown_60|gray_70;'\
+                   'path=a:3,b:_0,track:narrow;path=a:4,b:_0;frame=color:#800080;'\
+                   'icon=image:1873/15_open,sticky:1,large:1',
+              %w[
+                H9
+              ] => 'city=revenue:30,slots:2;path=a:0,b:_0,track:narrow;'\
+                   'path=a:1,b:_0,track:narrow;path=a:4,b:_0,track:narrow;'\
+                   'icon=image:1873/SBH9_open,sticky:1,large:1',
+              %w[
+                H21
+              ] => 'path=a:2,b:5,terminal:1',
+              %w[
+                I2
+              ] => 'city=revenue:yellow_40|green_50|brown_80|gray_120;path=a:1,b:_0;'\
+                   'path=a:2,b:_0,track:narrow;path=a:4,b:_0;frame=color:#800080',
+              %w[
+                I4
+              ] => 'city=revenue:yellow_40|green_50|brown_80|gray_120;path=a:1,b:_0;'\
+                   'path=a:2,b:_0,track:narrow;path=a:5,b:_0;frame=color:#800080',
+              %w[
+                I18
+              ] => 'city=revenue:yellow_30|green_40|brown_60|gray_70;'\
+                   'path=a:2,b:_0,track:narrow;frame=color:#800080;'\
+                   'icon=image:1873/13_open,sticky:1,large:1',
+              %w[
+                J7
+              ] => 'city=revenue:yellow_60|green_80|brown_120|gray_180;path=a:1,b:_0;'\
+                   'path=a:3,b:_0,track:narrow;path=a:4,b:_0;frame=color:#800080',
+              # implicit tiles
+              %w[
+                C20
+              ] => 'path=a:2,b:5',
+              %w[
+                D21
+              ] => 'path=a:2,b:0',
+              %w[
+                F17
+              ] => 'path=a:1,b:4',
+              %w[
+                F19
+              ] => 'path=a:1,b:3;path=a:5,b:3',
+              %w[
+                H1
+              ] => 'path=a:5,b:3,track:narrow',
+              %w[
+                J5
+              ] => 'path=a:2,b:4',
+            },
+          }
+          aag_variant? ? hexes_aag : hexes_original
```

Comment:
Rather than repeating the entire map except for the F7 hex, you can do this:

In game hexes:

```
        %w[
                F7
              ] => 'city=revenue:20;city=revenue:20;path=a:1,b:_0,track:narrow;'\
                   "path=a:3,b:_1,track:narrow;upgrade=cost:#{aag_variant? ? 100 : 50},terrain:mountain;"\
                   'icon=image:1873/11_open,sticky:1,large:1',
```

Note the double quotes on that line. Nothing else needs to change for game_hexes.

Reply:
This is one of the reasons I like to put the map definition in the game_hexes method rather than the constant HEXES. You can evaluate game options in that method whereas you cannot in HEXES.

Reply:
This is the elegance I was looking for! Fixed. Want to check it out again @roseundy ?